src/mlpack/tests/sumtree_test.cpp

```diff
@@ -13,82 +13,78 @@
 
 #include <mlpack/methods/reinforcement_learning/replay/sumtree.hpp>
 
-#include <boost/test/unit_test.hpp>
-#include "test_tools.hpp"
+#include "catch.hpp"
+#include "test_catch_tools.hpp"
 
 using namespace mlpack;
 using namespace mlpack::rl;
 
-BOOST_AUTO_TEST_SUITE(SumTreeTest);
-
 /**
  * Test that we set the element.
  */
-BOOST_AUTO_TEST_CASE(SetElement)
+TEST_CASE("SetElement", "[SumTreeTest]")
 {
   SumTree<double> sumtree(4);
   sumtree.Set(0, 1.0);
   sumtree.Set(1, 0.8);
   sumtree.Set(2, 0.6);
   sumtree.Set(3, 0.4);
 
-  BOOST_CHECK_CLOSE(sumtree.Sum(), 2.8, 1e-8);
-  BOOST_CHECK_CLOSE(sumtree.Sum(0, 1), 1.0, 1e-8);
-  BOOST_CHECK_CLOSE(sumtree.Sum(0, 3), 2.4, 1e-8);
-  BOOST_CHECK_CLOSE(sumtree.Sum(1, 4), 1.8, 1e-8);
+  CHECK(sumtree.Sum() == Approx(2.8).epsilon(1e-10));
+  CHECK(sumtree.Sum(0, 1) == Approx(1.0).epsilon(1e-10));
+  CHECK(sumtree.Sum(0, 3) == Approx(2.4).epsilon(1e-10));
+  CHECK(sumtree.Sum(1, 4) == Approx(1.8).epsilon(1e-10));
```

Comment:
I'd be fine changing these to `REQUIRE` for consistency, but it doesn't make much difference either way. :+1: